lib/puppet/functions/azure_key_vault/secret.rb

```diff
@@ -4,33 +4,58 @@
 Puppet::Functions.create_function(:'azure_key_vault::secret', Puppet::Functions::InternalFunction) do
   # @param vault_name Name of the vault in your Azure subcription.
   # @param secret_name Name of the secret to be retrieved.
-  # @param api_versions_hash A Hash of the exact versions of the metadata_api_version and vault_api_version to use.
```

Comment:
Instead of adding a new parameter, included new keys in `api_versions_hash`, and renamed it as api_endpoint_hash

Reply:
Hey @kev-in-shu ,

I was thinking of something along the lines of this.

## Service Principal

ruby
```
$important_secret = azure_key_vault::secret('production-vault', 'important-secret', {
  vault_api_version    => '2016-10-01',
  service_principal_credentials => {
    tenant_id     => '00000000-0000-1234-1234-000000000000',
    client_id     => '00000000-0000-1234-1234-000000000000',
    client_secret => Sensitive('00000000-0000-1234-1234-000000000000'),
  }
})
```

## MSI


ruby
```
$important_secret = azure_key_vault::secret('production-vault', 'important-secret', {
  vault_api_version    => '2016-10-01',
  metadata_api_version => '2018-04-02'
})
```

This would make it easy to understand what mode of authentication is being used ( either MSI or service_principal ).

Reply:
Hi @TraGicCode,

I just pushed some new code to match your suggestion.

Kevin